### PR TITLE
Added basic unit tests

### PR DESCRIPTION
**Pull request by sufisaid**
Added tests for
- Fetching ip info
- Saving ip info to db
- Throwing exception on duplicate ip

Added TestUtil with test data and test tools.

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const {getIPInfo} = require('../lib/ipinfo');
+const database = require('../lib/database');
+const TestUtil = require('./test-util');
+const assert = require('assert');
+
+const ipInfoTestData = TestUtil.getIPInfoTestData();
+
+describe('IP Info', function () {
```
Comment (KonradLinkowski) [inline test/test.js:8]: Could you use arrow function instead?

File: test/test-util.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    getRandomIP: () => {
+        let ip = "";
+        for(let i = 0; i < 4; i++) {
+            ip += Math.floor(Math.random() * 255) + 1 + ".";
+        }
+        return ip.slice(0,-1);
+    },
+
+    getIPInfoTestData: () => {
+        return {
```
Comment (KonradLinkowski) [inline test/test-util.js:14]: You could use [faker](https://www.npmjs.com/package/faker) for this one, but it's out of scope of the task.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+            const ip = "186.12.170.197";
+            const result = await getIPInfo(ip)
+
+            assert.strictEqual(result.ip, ip,"IP info not fetched");
+        });
+    });
+
+    describe('addEntry', function () {
+        it('should add new entry to database', async function () {
+            let result = await database.addEntry(ipInfoTestData);
```
Comment (KonradLinkowski) [inline test/test.js:21]: `const`

File: test/test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            assert.strictEqual(result.ip, ipInfoTestData.ip ,"IP Info saved");
+        })
+    })
+
+    describe('addEntry', function () {
+        it('should throw duplicate key exception', async function () {
```
Comment (KonradLinkowski) [inline test/test.js:27]: You can move this `it` to the describe above.

File: test/test.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+            await assert.rejects(async () => {
```
Comment (KonradLinkowski) [inline test/test.js:28]: This is cool. I didn't see that construction before :+1:

**Review by KonradLinkowski — CHANGES_REQUESTED** [review]
Looks good, just do the minor changes.

File: test/test.js (hunk 1 above)
Comment (sufisaid) [reply test/test.js:8]: Arrow functions shouldn't be used as they would change the context and 'this' would not be bind to Mocha.

File: test/test-util.js (hunk 2 above)
Comment (sufisaid) [reply test/test-util.js:14]: Yes, faker would be good option, and util is not a good name in this case, but let’s leave that for another time :)

File: test/test.js (hunk 5 above)
Comment (sufisaid) [reply test/test.js:28]: Yea, it's a bit specific due to function type and what I wanted to test/catch.

File: test/test.js (hunk 4 above)
Comment (sufisaid) [reply test/test.js:27]: Will do.

File: test/test.js (hunk 3 above)
Comment (sufisaid) [reply test/test.js]: Done

File: test/test.js (hunk 4 above)
Comment (sufisaid) [reply test/test.js:25]: Done

**Review by KonradLinkowski — APPROVED** [review]

**Pull request merged**